src/meta/src/hummock/compaction/picker/tier_compaction_picker.rs

```diff
@@ -408,12 +309,14 @@ impl TierCompactionPicker {
             // levels to ensure that we can merge as many sub_levels as possible
             let tier_sub_level_compact_level_count =
                 self.config.level0_overlapping_sub_level_compact_level_count as usize;
-            if select_level_inputs.len() < tier_sub_level_compact_level_count
-                && waiting_enough_files
-            {
+            let min_level0_compact_file_number = self.config.level0_tier_compact_file_number
```

Comment:
This seems more reasonable than before, can we use it in base level compaction as well?